### ISSUE TITLE
Settle on precise terminology related to nicknames and petnames

### ISSUE THREAD

**Issue by cdata** [subconsciousnetwork/noosphere#563]
The Noosphere name system is designed after a "petname"-style name system. The term petname is borrowed from [this article](http://www.skyhunter.com/marcs/petnames/IntroPetNames.html), which distinguishes petnames and nicknames:

- **Nicknames:** "can be used to assist in discovery of keys, and for help in selecting a petname. Nicknames are chosen by the owners of  keys in hopes of creating a distinctive, if not unique, mapping from the memorable nickname to the key."
- **Petnames:** "are our private bidirectional references to keys. There are many Mark Millers, but there is one specific Mark Miller that the name means to me, the Mark Miller who works with object-capabilities for secure cooperation. "Mark Miller" is Mark Miller's nickname; it also happens to be my petname for the same individual."

However, "pet name" as a matter of contemporary definition carries a slightly inflected meaning:

> pet name
> _noun_: a name that a person uses for someone to show love or affection
> _[Merriam Webster's Dictionary](https://www.merriam-webster.com/dictionary/pet%20name)_

Whereas nickname is very close to our intended meaning for petname:

> nickname
> _noun_
> 1: a usually descriptive name given instead of or in addition to the one belonging to a person, place, or thing
> 2: a familiar form of a proper name (as of a person or a city)
> _[Merriam Webster's Dictionary](https://www.merriam-webster.com/dictionary/nickname)_

Given this distinction, we have begun preferring the term "nickname" in Subconscious UI, and as of #530 we have also introduced it to end-user-facing aspects of the Noosphere CLI.

All that having been said, we haven't done much in the way to document our rationalization for this, or the rules to follow when deciding how to use such terms. And, we are also faced with the problem that other things in the Noosphere domain could rightly be described as nicknames, such as:

- The human-readable name assigned by the user to a device key
- The human-readable name assigned by a user to an authorization
- ...and perhaps other things in the future

So, this issue calls us to discuss this topic and land on precise definitions for terms such as petname and nickname (and/or other related terms), including prescriptions for when to use each.

**Comment by gordonbrander** [comment]
The original petname paper calls names that you suggest for yourself or others "nicknames". Worth noting that papers from Spritely use the term "proposed name" and "self-proposed name" for this concept https://spritely.institute/static/papers/petnames.html

**Comment by gordonbrander** [comment]
My feeling is that "petname" is not a great name, but is pretty well cemented as a term-of-art in decentralized systems. I think it would probably confuse things if we did not call our name system a "petname-based name system".

However, I do not think that petname is a good user-facing term. I feel that we should not use it in user-facing copy, any more than we would use "DID" or "packet" in user-facing contexts. These are implementation details that mean something to implementors, but not to users.

Better user-facing terms might be:

- Nickname
- Handle
- Or maybe just "Name"... after all, the iPhone phone book doesn't use a special term.

**Comment by cdata** [comment]
Thanks for the extra detail! So, the full list of terms may have just grown to include:

- Petname
- Nickname
- Handle
- Proposed name
- Name

My hope is that we can land on precise definitions / prescriptions for (or against) usage for _all_ such terms.

**Comment by gordonbrander** [comment]
Another note: When you give someone else a name on Discord, they call this a "nickname".

**Comment by cdata** [comment]
> This is a potential area for abuse, which I suspect y'all are already thinking about. (e.g. if I petname someone as "ssn:22392739" or "big-ol-racist")

Yes, abuse vectors here are real and very serious. As a base case, you don't see someone's petname for you just because they follow you, but you can still encounter it via a triangular (or yet-more complex) traversal path. The remedy for this problem will have to be multi-dimensional and include pro-active filtering / warning within client applications such as Subconscious, as well as T&S protocols/countermeasures implemented by service providers, among other things.
